modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotChecker.java

```diff
@@ -0,0 +1,698 @@
+
+import static org.apache.ignite.internal.pagemem.PageIdAllocator.FLAG_DATA;
+import static org.apache.ignite.internal.pagemem.PageIdAllocator.FLAG_IDX;
+import static org.apache.ignite.internal.pagemem.PageIdAllocator.INDEX_PARTITION;
+import static org.apache.ignite.internal.processors.cache.distributed.dht.topology.GridDhtPartitionState.OWNING;
+import static org.apache.ignite.internal.processors.cache.distributed.dht.topology.GridDhtPartitionState.fromOrdinal;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.CACHE_DATA_FILENAME;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.FILE_SUFFIX;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.ZIP_SUFFIX;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.cacheDirectories;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.cacheGroupName;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.cachePartitionFiles;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.partId;
+import static org.apache.ignite.internal.processors.cache.persistence.partstate.GroupPartitionId.getTypeByPartId;
+import static org.apache.ignite.internal.processors.cache.persistence.snapshot.IgniteSnapshotManager.SNAPSHOT_METAFILE_EXT;
+import static org.apache.ignite.internal.processors.cache.persistence.snapshot.IgniteSnapshotManager.databaseRelativePath;
+import static org.apache.ignite.internal.processors.cache.persistence.snapshot.dump.CreateDumpFutureTask.DUMP_FILE_EXT;
+import static org.apache.ignite.internal.processors.cache.persistence.wal.reader.StandaloneGridKernalContext.closeAllComponents;
+import static org.apache.ignite.internal.processors.cache.persistence.wal.reader.StandaloneGridKernalContext.startAllComponents;
+import static org.apache.ignite.internal.processors.cache.verify.IdleVerifyUtility.calculatePartitionHash;
+import static org.apache.ignite.internal.processors.cache.verify.IdleVerifyUtility.checkPartitionsPageCrcSum;
+
+/** */
+public class SnapshotChecker {
+    /** */
+    protected final IgniteLogger log;
+
+    /** */
+    @Nullable protected final GridKernalContext kctx;
+
+    /** */
+    protected final Marshaller marshaller;
+
+    /** */
+    @Nullable protected final ClassLoader marshallerClsLdr;
+
+    /** */
+    protected final EncryptionSpi encryptionSpi;
+
+    /** */
+    @Nullable protected final CompressionProcessor compression;
+
+    /** */
+    protected final ExecutorService executor;
+
+    /** */
+    public SnapshotChecker(
+        GridKernalContext kctx,
+        Marshaller marshaller,
+        ExecutorService executorSrvc,
+        @Nullable ClassLoader marshallerClsLdr
+    ) {
+        this.kctx = kctx;
+
+        this.marshaller = marshaller;
```

Comment:
It looks like `marshaller` and `marshallerClsLdr` is used only in single method. Should we move then this params from constructor to those methods?

Reply:
Not sure. This single method is used in other `read-smth` methods like `readSnapshotMetadata()` where the params would be required too. Looks unhandy.